src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Api.Enduser/Services/EnduserConnectionService.cs

```diff
@@ -211,19 +211,23 @@ private async Task<Result<AssignmentPackage>> AddPackage(Guid fromId, Guid toId,
             assignment = existingAssignments.First();
         }
 
+        var check = await CheckPackage(fromId, packageIds: [fromId], cancellationToken);
+        if (check.IsProblem)
+        {
+            return check.Problem;
+        }
+
         var userPackageFilter = ConnectionPackageRepository.CreateFilterBuilder()
             .Equal(t => t.ToId, DbAudit.Value.ChangedBy)
             .Equal(t => t.FromId, fromId)
             .Equal(t => t.PackageId, packageId);
 
-        var userPackags = await ConnectionPackageRepository.GetExtended(userPackageFilter, callerName: SpanName("Get extended packages assignments"), cancellationToken: cancellationToken);
-        var userpackage = userPackags.Where(p => p.PackageId == packageId)
+        var userPackages = await ConnectionPackageRepository.GetExtended(userPackageFilter, callerName: SpanName("Get extended packages assignments"), cancellationToken: cancellationToken);
```

Comment:
I stedet for å gjøre ekstra oppslag her burde vel egentlig PackageIsAssignableToRecipient skrives om til å sjekke basert på responsen fra DelegationCheck.
Sjekken er jo bare at en gitt urn ikke er lov å delegere dersom mottaker er en organisasjon, og urn er del av delegationcheck responsen.

Reply:
mm, true true. litt tungt